Add paragraph drop cap style configuration

pbking opened the pull request:
Adds configuration of the drop cap style.  This should complete the necessary changes for the "paragraph" block portion of #3413

Before:
![image](https://user-images.githubusercontent.com/146530/110532251-94acfe80-80ea-11eb-9cc5-49c42f5e7c6e.png)

After:
![image](https://user-images.githubusercontent.com/146530/110532143-79da8a00-80ea-11eb-9ed9-cae44a8c7aaf.png)

pbking committed:
Add paragraph drop cap style configuration

jffng commented on blank-canvas-blocks/experimental-theme.json:
```diff
@@ -171,6 +171,15 @@
+				"paragraph": {
+					"dropcap": {
+						"typography": {
+							"font-family": "var(--wp--preset--font-family--base)",
```
Nit pick but are we going with camel case or trying matching the CSS attribute?

I saw in [another PR a camel case](https://github.com/Automattic/themes/pull/3437/files#diff-6b32b96741b2b4b9e621f53f3709cb5e5dcbb41c0467f83a2f30c99577f724b8R149), and it made me think we should be consistent with what Gutenberg is doing.

MaggieCabrera replied:
+1 on this, I opened a PR to change to camel case here https://github.com/Automattic/themes/pull/3444

jffng commented:
Nice one, this is close but I'm wondering if we can get it closer, at least align the top of the letter with the rest of the paragraph. 

BCB this PR | Blank Canvas
------- | -------
<img width="645" alt="Screen Shot 2021-03-09 at 3 23 44 PM" src="https://user-images.githubusercontent.com/5375500/110552098-8d3a2500-80eb-11eb-9954-6bc8a19e7197.png"> | <img width="647" alt="Screen Shot 2021-03-09 at 3 23 39 PM" src="https://user-images.githubusercontent.com/5375500/110552104-90351580-80eb-11eb-93c7-8888f9a0ac20.png">

pbking committed:
Added margin and adjusted values of drop cap

pbking commented:
Added a (single) "margin" property to apply to the dropcap letter.  That let's us pull it down to align it with the top.
Adjusted the font-size and font-weight to match Blank Canvas.  
Then I further increased the font-size (a little) so that it would fit a LITTLE more snugly.  

Blank Canvas Numbers:
![image](https://user-images.githubusercontent.com/146530/110640856-51e93600-817f-11eb-9fb9-26ed1caa1b96.png)

Final: 
![image](https://user-images.githubusercontent.com/146530/110640564-f1f28f80-817e-11eb-9744-f4cad7e60828.png)
![image](https://user-images.githubusercontent.com/146530/110640620-03d43280-817f-11eb-80a3-cec96861aa95.png)

pbking committed:
Refactored kebab variables to camelCase for dropcap

jffng committed:
Spaces to tab.

jffng commented:
This looks much better, thank you! 🚢
